File: frontend/app/Models/Shop.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Shop extends Model
{
    use HasFactory;

    protected $fillable = [
        'user_id',
        'shop_name',
        'city',
        'address',
        'phone_number',
        'immage',
    ];

    public function user() {
        return $this->hasOne('App\Models\User', 'id', 'user_id');
    }
}

File: backend/models/shop.go
package models

import (
    "time"
    // "gorm.io/gorm"
)

type Shop struct {
    ID          uint           `gorm:"primaryKey"`
    UserID      uint
    ShopName    string         `gorm:"size:255"`
    City        string         `gorm:"size:255"`
    Address     string         `gorm:"size:255"`
    PhoneNumber string         `gorm:"size:255"`
    Image       string         `gorm:"size:255;default:'default.jpg'"`
    CreatedAt   time.Time
    UpdatedAt   time.Time
}

func (Shop) TableName() string {
    return "shops"
}
